File: problema 22/c-languaje/problema_22.c
#include <stdio.h>
#include <stdlib.h>

void sort(int arr[], int n, int ascending) {
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if ((ascending && arr[j] > arr[j + 1]) || (!ascending && arr[j] < arr[j + 1])) {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}

void printArray(int arr[], int n) {
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

int main() {
    int n;
    printf("Ingrese el tamaño del arreglo (entre 20 y 50): ");
    scanf("%d", &n);

    if (n < 20 || n > 50) {
        printf("Error: el arreglo debe ser de entre 20 y 50 elementos.\n");
        return 1;
    }

    int *arr = (int *)malloc(n * sizeof(int));
    printf("Ingrese %d elementos:\n", n);
    for (int i = 0; i < n; i++) {
        printf("Elemento %d: ", i + 1);
        scanf("%d", &arr[i]);
    }

    sort(arr, n, 1); // Sort ascending
    printf("Arreglo ordenado de menor a mayor:\n");
    printArray(arr, n);

    sort(arr, n, 0); // Sort descending
    printf("Arreglo ordenado de mayor a menor:\n");
    printArray(arr, n);

    free(arr);
    return 0;
}

File: problema 22/fortran/problema_22.f90
program sort_array
    implicit none
    integer, allocatable :: arr(:)
    integer :: n, i, j, temp

    print *, 'Ingrese el tamaño del arreglo (entre 20 y 50):'
    read *, n

    if (n < 20 .or. n > 50) then
        print *, 'Error: el arreglo debe ser de entre 20 y 50 elementos.'
        stop
    endif

    allocate(arr(n))

    print *, 'Ingrese ', n, ' elementos:'
    do i = 1, n
        print *, 'Elemento ', i, ':'
        read *, arr(i)
    end do

    ! Ordenar de menor a mayor
    do i = 1, n - 1
        do j = 1, n - i
            if (arr(j) > arr(j + 1)) then
                temp = arr(j)
                arr(j) = arr(j + 1)
                arr(j + 1) = temp
            endif
        end do
    end do

    print *, 'Arreglo ordenado de menor a mayor:'
    print *, arr

    ! Ordenar de mayor a menor
    do i = 1, n - 1
        do j = 1, n - i
            if (arr(j) < arr(j + 1)) then
                temp = arr(j)
                arr(j) = arr(j + 1)
                arr(j + 1) = temp
            endif
        end do
    end do

    print *, 'Arreglo ordenado de mayor a menor:'
    print *, arr

    deallocate(arr)
end program sort_array